pulp_docker/app/tasks/tag.py

```diff
@@ -0,0 +1,39 @@
+from pulpcore.plugin.models import Repository, RepositoryVersion, ContentArtifact
+from pulp_docker.app.models import ManifestTag, Manifest
+
+
+def create_new_repository_version(manifest_pk, tag, repository_pk):
+    """
+    Create a new repository version out of the passed tag name and the manifest.
+
+    If the tag name is already associated with an existing manifest with the same digest,
+    no new content is created. Note that a same tag name cannot be used for two different
+    manifests. Due to this fact, an old ManifestTag object is going to be removed from
+    a new repository version when a manifest contains a digest which is not equal to the
+    digest passed with POST request.
+    """
+    manifest = Manifest.objects.get(pk=manifest_pk)
+    artifact = manifest._artifacts.all()[0]
+
+    existing_manifest_tag_list = ManifestTag.objects.filter(name=tag).exclude(
+        tagged_manifest=manifest
+    )
+
+    manifest_tag, _ = ManifestTag.objects.get_or_create(
```

Comment:
in case we create a new tag, can we add it to the created_resources?

Reply:
https://github.com/pulp/pulp_rpm/commit/120b8f29cbdecae74cd4990c659522ad24734e54#diff-e19a9590ea08dbcd574065d3ab6d4123R37